build/webpack.config.stats.ts

```diff
@@ -0,0 +1,98 @@
+import * as CleanWebpackPlugin from 'clean-webpack-plugin'
+import * as fs from 'fs'
+import * as path from 'path'
+// import * as _ from 'lodash'
+import * as webpack from 'webpack'
+import config from '../config'
+
+// const pkg = require('../package.json')
+const { paths } = config
+
+// Interfaces no longer exist after transpiling, causing webpack to throw errors about them.
+// ERROR:    https://github.com/webpack/webpack/issues/7378
+// HACK FIX: https://github.com/TypeStrong/ts-loader/issues/653
+class IgnoreNotFoundExportPlugin {
+  apply(compiler) {
+    const messageRegExp = /export '.*'( \(reexported as '.*'\))? was not found in/
+
+    const doneHook = stats => {
+      stats.compilation.warnings = stats.compilation.warnings.filter(
+        warn => !(warn && messageRegExp.test(warn.message)),
+      )
+    }
+
+    if (compiler.hooks) {
+      compiler.hooks.done.tap('IgnoreNotFoundExportPlugin', doneHook)
+    } else {
+      compiler.plugin('done', doneHook)
+    }
+  }
+}
+
+const makeConfig = (srcPath, name) => ({
+  mode: 'production',
+  name: 'client',
+  target: 'web',
+  entry: paths.dist(path.join('es', srcPath)),
+  output: {
+    filename: `${name}.js`,
+    path: paths.base('stats'),
+    pathinfo: true,
+    publicPath: config.compiler_public_path,
+  },
+  devtool: false,
+  module: {
+    rules: [
+      {
+        test: /\.(js|ts|tsx)$/,
+        loader: 'ts-loader',
+        include: /src/,
+        options: {
+          configFile: paths.base('build/tsconfig.es.json'),
+          transpileOnly: true,
+          onlyCompileBundledFiles: true,
+          compilerOptions: {
+            declaration: false,
+          },
+        },
+      },
+    ],
+  },
+  externals: {
+    react: 'react',
+    'react-dom': 'reactDOM',
+  },
+  plugins: [
+    new CleanWebpackPlugin([paths.base('stats')], {
+      root: paths.base(),
+      verbose: false, // do not log
+    }),
```

Comment:
Do we really need this? As I see we already have a `rimraf` as dependency and we can use it as a part of Gulp's task.